packages/orchestration/src/exos/local-orchestration-account.js

```diff
@@ -143,27 +147,28 @@ export const prepareLocalOrchestrationAccountKit = (
          * @param {string} validatorAddress
          * @param {Amount<'nat'>} ertpAmount
          */
-        async Delegate(validatorAddress, ertpAmount) {
+        Delegate(validatorAddress, ertpAmount) {
           trace('Delegate', validatorAddress, ertpAmount);
 
-          return zcf.makeInvitation(async seat => {
-            // TODO should it allow delegating more BLD?
+          return zcf.makeInvitation(seat => {
             seat.exit();
-            return this.facets.holder.delegate(validatorAddress, ertpAmount);
+            return watch(
+              this.facets.holder.delegate(validatorAddress, ertpAmount),
+            );
           }, 'Delegate');
         },
```

Comment:
What is the correct way to return an invitation here? I think we only need the offer handler to be resumable, which I've done here. This, or perhaps something else, seems to result in errors being swallowed.

Reply:
This seems to be correct. The issue faced is due to https://github.com/Agoric/agoric-sdk/issues/9308